test/functional/legacy/eval_spec.lua

```diff
@@ -495,6 +495,18 @@ describe('eval', function()
       ' abcE4b10-4\000abcE4b10-4-2')
   end)
 
+  it("setreg('0',getreg('a',1,1)) doesn't crash for unset register", function()
```

Comment:
Please use `v:_null_list`, not `getreg()`. This way you need a single line here: `eq(0, eval('setreg("0", v:_null_list)'))`: if Neovim crashed `eval` would return `nil` and not `0`.